src/main/java/jp/co/soramitsu/sora/json/Flattener.java

```diff
@@ -0,0 +1,172 @@
+package jp.co.soramitsu.sora.json;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.val;
+import lombok.var;
+
+
+public class Flattener {
+
+  private int features = 0;
+  private char dictkey = '/';
+  private char arrkey = '_';
+  private ObjectMapper mapper = new ObjectMapper();
+
+  private void flatten(JsonNode root, String path, ObjectNode out) {
+    if (root.isObject()) {
+      if (root.size() == 0) {
+        // it is empty object
+        out.set(path, root);
+        return;
+      }
+
+      val fields = root.fields();
+      while (fields.hasNext()) {
+        var it = fields.next();
+        var key = sanitize(it.getKey());
+
+        flatten(it.getValue(), path + dictkey + key, out);
+      }
+    } else if (root.isArray()) {
+      if (root.size() == 0) {
+        // it is empty array
+        out.set(path, root);
+        return;
+      }
+
+      val fields = root.iterator();
+      long counter = 0;
+      while (fields.hasNext()) {
+        var it = fields.next();
+        flatten(it, path + arrkey + (counter++), out);
+      }
+
+    } else if (root.isValueNode()) {
+      out.set(path, root);
+
+    } else {
+      throw new FlattenerException("type is neigher object, array or value");
+    }
+  }
+
+  private void deflatten(JsonNode root, ObjectNode out) {
+    throw new RuntimeException("not implemented");
```

Comment:
I suggest changing to `UnsupportedOperationException`
Or better: having specific exception
```java
public class DeflattenNotSupported extends RuntimeException {
  public DeflattenNotSupported() {
    super("Deflatten not supported!");
  }
}
```

Reply:
It is good to have separate exception, but this method will be implemented soon, so there is no need in a separate class.